### ISSUE TITLE
golem error

### ISSUE THREAD

**Issue by Shibva** [WayofTime/BloodMagic#1544]
Hello Im aware that this is not following the template 
but I would like to notify you that after some version testing that Your Mod Blood Magic (2.4.0-102) is Causing an iterfearence to the Golems From Thaumcraft to Unabe to fly correctly due to recent changes.

please note that this wasent an issue until versions AFTER version 2.3,3-101
it was version 2.3.3-101 of the mod that the golems from Thaumcraft with there Flight upgrade were working correctly

Id hope that you can fix this issue 
I really like this mod lots

**Comment by keraldi** [comment]
The template is there for a reason.
Unless you are not talking about just blood magic and still telling us version numbers and items you cannot expect us to know anything about what you're talking.

Fill out the issue template.
Provide the mod versions you tested it with which break the expected behavior (I assume it worked in BM-101 but broke in BM-102 with some Thaumcraft version (we need the thaumcraft version).

Verify that the issue exists even if you have only a minimum amount of mods (i.e. Thaumcraft and Bloodmagic and their dependencies).

Keep in mind that coremods can break stuff with a much higher probability than normal mods.

Give us the Thaumcraft version.
And for all sakes, provide a step-by-step guide to replicate the issue, that includes:

1. Names of all Items/mechanics used
2. how to click where, when and under which circumstances
3. ...

If you do not adhere to the issue template, the issue WILL be ignored, especially if it is an assumed or supposed cross-mod issue.

**Comment by Shibva** [comment]
#### What happens: golems from Thaumcraft do Not Fly 



#### What you expected to happen: For golems to fly like the should of before i updated ONLY Blood Magic



#### Steps to reproduce:
Install 
[s-1.0.zip](https://github.com/WayofTime/BloodMagic/files/2871391/s-1.0.zip) via twich

2 load game 
3 make a world
4 Create a Golem with the arcrane levitator upgrade
5 Place Golem at any height (recommend 4 blocks or higher to see results)

if the golem moves and falls when it should be levitating then that is the problem

...

____
#### Affected Versions (Do *not* use "latest"):

- BloodMagic: 2.4.0-102
- Minecraft: 1.12.2
- Forge: 14.23.5.2807

**Comment by wrincewind** [comment]
There's a lot of mods in that modlist! have you tried disabling every mod aside from Blood Magic and Thaumcraft and trying this again? (In a new world, obviously)

**Comment by Shibva** [comment]
Ive have determend this

when I update everything Exept Blood Magic It was working Perfectly
it was only after updating Blood Magic to 102 that the issue arose
and yes this has been tested

**Comment by Shibva** [comment]
it should also be noted that Mods such as OptiFine and Vannila Fix have been ruled out

**Comment by wrincewind** [comment]
ok, it's definitely happening. Golems are plummeting to the ground, and when you give them a task in the air, they 'float' on the ground, looking plaintively up at their goal. It actually makes me kind of sad. 

Forge: 2807
Baubles 1.5.2 
guide-api 2.1.8-63
Blood Magic 102
Thaumcraft BETA26

No other mods present, default configs, etc.

**Comment by Shibva** [comment]
good to know that my asupton was correct

**Comment by Shibva** [comment]
oh and please also note that (an old issue that is probaly no longer an issue) that the game crashes when the Alter Ticks due to the Embers liquid pumps are used to add or drain the alter.
But this can be a Minnor inconvinence

**Comment by Shibva** [comment]
Also when i Reverted my Blood magic version to 101 It stated some features were missing such as Somthing to do with Levitating or somthing of that sort

its probaly safe to say that adding a new feature with Levitating off the ground is probaly what is causing the issue

thankyou for your time and I hope that This can be Fixed asap
(btw i just recreated a pack i sent you for convience just for this test and it was preset just like on my main Modpack, Thank me when you hear that 305 Mods are present with the original and that the probelm was able to be recreated wiht the mods provided in the experimental pack)

**Comment by keraldi** [comment]
No idea how to create a golem with levitator upgrade, cannot reproduce the issue because the instructions are incomplete.

**Comment by Shibva** [comment]
Omg

Is SImple
Just type in  the command
/thaumcraft research @p all
then buld the golem press multistructure as shown in the Thaumnomicon
then use the componets to make a golem

**Comment by keraldi** [comment]
... I've got better things to do than research how to get something specific when there's an issue template that's made specifically to tell the contributors how exactly you replicated your issue/bug.

On the other hand, if it is so simple to do the 10 or so not jumping into your eye steps to replicate the issue, so please tell me how and why people flood the discord server asking questions about text appearing when starting up modded Minecraft telling them that mods are missing.

I contribute to the Blood Magic mods. I have other stuff to do. If you expect me to research stuff I don't regularly work with and/or not part of the mod and be unfriendly about it on top of it, you should not expect it to be fixed at all.

While you do your mindless gaming and complaining in your free time, I try to use my free time for the betterment of yours.

Nobody here is getting paid for modding in a manner that would be reimbursement for the harassment of people like you.

That being said, I wouldn't even work for you under such circumstances even if you would pay me.

That being said, I was aware of the command and used it (as it has been in previous versions) but the Golem Workstation is something completely new and something I have never used before.
Point is: the issue template that you initially refused to fill out has instructions for a step-by-step reproduction of the bug which means STEP BY STEP (which you have not provided).

Step by step means that anyone with minimal knowledge about the project (in this case, Minecraft and Blood Magic) would be able to reproduce your issue.
The reason for all this is to bring a certain standard to issues and make working them as efficient as possible.

That was not the case. That would already warrant closing this issue or ignoring it.

Anyways. I'll give you one last chance to provide a reasonable explaination and step by step procedure to make working the issue in my limited timeframe possible.

Otherwise I'll ignore it at least until I have an extended period of time to experiment on it (which might never happen (and you can include that I'll want to not work the issue at all because you were unreasonable)).

**Comment by kingpotato2000** [comment]
Have tested with just Blood Magic 102 and Thaumcraft. Golem just refuses to fly, its just walks off the edge

**Comment by kingpotato2000** [comment]
Tested again with Blood Magic disabled, golem works as intended

**Comment by kingpotato2000** [comment]
Tested once again with Blood Magic 101, golem works as normal

**Comment by Shibva** [comment]
thankyou
hope this gets fixed soon

**Comment by kingpotato2000** [comment]
soon as someone works out what is causing the issue

**Comment by TheMadmanofChaos** [comment]
*Sigh* I feel bad for you Tobias. Allow me to at least try and help with this current head ache.

#### Issue Description:
Version 1.12.2-2.4.0-102 of Blood Magic is conflicting with mod "Thaumcraft", version tested being 1.12.2-6.1BETA26, preventing golems with the upgrade "Golem Mobility: Flight" from, well, flying. 

Also, note: No obvious errors are thrown out into the client's console. Sadly isn't that easy.
#### What happens:
In all cases tested, the golem is earth bound when it should be flying, free as a... Well, an enslaved minion with a jetpack instead of legs. 


#### What you expected to happen:
Said minions being able to hover over short distances, like a discount Superman.


#### Steps to reproduce:

1. Create new world.
2. Run command "/thaumcraft research @p all" to unlock all of thaumcraft.
3. Wait a minute for all the unlock messages to go away.
4.Build the golem press, outlined in Thaumonomicon entry "Clockwork Minds", under the golem tab. A screen shot will be included in this post as a reference point. Once all the blocks in in the correct locations, right click the structure with item "Salus Mundus" in order to form the structure.

![2019-02-16_17 19 08](https://user-images.githubusercontent.com/5651102/52906592-a5912b80-3214-11e9-92fd-e2148332564b.png)


5.Right click the thing in order to open the golem building interface. Use the arrow buttons of the interface to switch the legs slot to the "Levitation Module". All the other settings will function in the default state.

![2019-02-16_17 24 04](https://user-images.githubusercontent.com/5651102/52906599-ba6dbf00-3214-11e9-9dd7-84663e922cab.png)


6.Gather all needed items listed in the upper right section of the interface, then click the build button in the lower right. The essentia can be provided using a warded jar, an essentia tube, and a phial of machina essentia. Screenshot provided to aid in this process. Just right click the phial on the jar in order to dump it inside.

![2019-02-16_17 24 09](https://user-images.githubusercontent.com/5651102/52906602-c48fbd80-3214-11e9-896f-16e18565f15e.png)


7.With newly formed golem helper minion, build a small raised platform, place a chest on top of it. Again, screen shot example is provided.

![2019-02-16_18 02 21](https://user-images.githubusercontent.com/5651102/52906625-0a4c8600-3215-11e9-9bce-53532bba5d68.png)


8.On this chest, place "Golem core: Gather" and "Golem Core: Fill". These will tell the golem what to do. The last screen shot should cover this part.
9.Right click on the "Gather" core with item "Golemancer's Bell" to open it's interface. Up all the range settings to max.
10.Place the golem down on your raised platform. 
11. Build out a second platform within a few blocks, ensuring there is a gab between them as seen in the earlier screenshots.
12. Drop an item down on this second platform, and watch as the helper minion falls down, See the shame on it's face, unable to perform it's assigned task.

____
#### Affected Versions (Do *not* use "latest"):

- BloodMagic: 1.12.2-2.4.0-102
- Minecraft: 1.12.2
- Forge: 14.23.5.2815
-Thaumcraft: 1.12.2-6.1BETA26,

If any further information is required, feel free to ask. I will do what is needed to aid in the resolving of this error.

**Comment by TehNut** [comment]
Long time, no see Madman 👋 

Thanks for the detailed explanation of how we can get to a point where we can even test 👍

**Comment by TheMadmanofChaos** [comment]
I arrive when needed. I wait. I watch. I act.

**Comment by Lord-Felidae** [comment]
I’d hate to barge in, but I would like to inform everyone that this issue may only be fixed from one side.

First, Thaumcraft is owned by the legendary @Azanor, and the latest version is closed source.

Second, sadly, Azanor has to leave the project and is currently discussing ownership with someone else.

If anyone can help, it would have to be Azanor. Otherwise this issue may not be resolved for quite some time.

**Comment by runescapejon** [comment]
I was doing some testing and figure i should put the result here i dont know if this can be helpful or not anyways.

without the bloodmagic i type 
`/summon armor_stand ~ ~ ~ {Invisible:1b,NoGravity:1b,Marker:1b,CustomName:"TEST",CustomNameVisible:1b}`

This command allow a floating text display but with bloodmagic installed it just fall down with no gravity of it being floating in the air. Furthermore i look in source and found the cause and the root of the issue it's here https://github.com/WayofTime/BloodMagic/blob/1.12/src/main/java/WayofTime/bloodmagic/potion/PotionEventHandlers.java#L96-L100 it seem that it's effecting other entities causing them to do weird and odd things.

**Comment by randombyte-developer** [comment]
Great that the root cause of the no-gravity issue had been figured out. I'm the dev if the holograms plugin and ofc it depends on the no-gravity nbt tag heavily. 

I've got various complaints from many mod pack servers.

**Comment by Shibva** [comment]
by chace where can i find the most recent version of Blood magic with this fix avalable

**Comment by keraldi** [comment]
You can compile the project as described in the project [Readme](../blob/1.12/README.md)
